services/commands-service.ts

```diff
@@ -72,8 +78,11 @@ export class CommandsService implements ICommandsService {
 				this.$logger.printMarkdown(~suggestionText.indexOf('%s') ? require('util').format(suggestionText, commandArguments) : suggestionText);
 			}
 
+			_.remove(this.commands, commandData => commandData.commandName === commandName);
```

Comment:
I think you can safely use `pop` here and below instead of `_.remove` because commands would be pushed one over the other just like a stack